test/GUI/test_gui.py

```diff
@@ -6,133 +6,114 @@
-    @patch('SeleniumLibrary.SelectElementKeywords.select_from_list_by_value')
-    def test_wait_for_and_select_from_list_by_value_simple(self, robot_call, select_list):
+    def test_wait_for_and_select_from_list_by_value_simple(self):
         mock_gui = Mock()
         GUILibrary.wait_for_and_select_from_list_by_value(mock_gui, "some_locator", "target")
-        assert robot_call.called_with(15, 1, "Set Focus To Element", "some_locator")
-        assert select_list.called_with("some_locator", "target")
+        mock_gui.wait_for_and_focus_on_element.assert_called_with("some_locator")
+        mock_gui.select_from_list_by_value.assert_called_with("some_locator", "target")
 
-    @patch('robot.libraries.BuiltIn.BuiltIn.wait_until_keyword_succeeds')
-    @patch('SeleniumLibrary.SelectElementKeywords.select_from_list_by_index')
-    def test_wait_for_and_select_from_list_by_index_simple(self, robot_call, select_list):
+    def test_wait_for_and_select_from_list_by_index_simple(self):
         mock_gui = Mock()
         GUILibrary.wait_for_and_select_from_list_by_index(mock_gui, "some_locator", "target")
-        assert robot_call.called_with(15, 1, "Set Focus To Element", "some_locator")
-        assert select_list.called_with("some_locator", "target")
+        mock_gui.wait_for_and_focus_on_element.assert_called_with("some_locator")
+        mock_gui.select_from_list_by_index.assert_called_with("some_locator", "target")
 
-    @patch('robot.libraries.BuiltIn.BuiltIn.wait_until_keyword_succeeds')
-    @patch('SeleniumLibrary.ElementKeywords.mouse_over')
-    def test_wait_for_mouse_over_simple(self, robot_call, mouse_over):
+    def test_wait_for_mouse_over_simple(self):
         mock_gui = Mock()
         GUILibrary.wait_for_and_mouse_over(mock_gui, "some_locator")
-        assert robot_call.called_with(15, 1, "Set Focus To Element", "some_locator")
-        assert mouse_over.called_with("some_locator")
+        mock_gui.wait_for_and_focus_on_element.assert_called_with("some_locator")
+        mock_gui.mouse_over.assert_called_with("some_locator")
 
-    @patch('robot.libraries.BuiltIn.BuiltIn.wait_until_keyword_succeeds')
-    @patch('SeleniumLibrary.ElementKeywords.mouse_over')
-    @patch('SeleniumLibrary.ElementKeywords.click_element')
-    def test_wait_for_mouse_over_and_click_simple(self, robot_call, mouse_over, click_element):
+    def test_wait_for_mouse_over_and_click_simple(self):
         mock_gui = Mock()
         GUILibrary.wait_for_and_mouse_over_and_click(mock_gui, "some_locator")
-        assert robot_call.called_with(15, 1, "Set Focus To Element", "some_locator")
-        assert mouse_over.called_with("some_locator")
-        assert click_element.called_with("some_locator")
+        mock_gui.wait_for_and_focus_on_element.assert_called_with("some_locator")
+        mock_gui.mouse_over.assert_called_with("some_locator")
+        mock_gui.click_element.assert_called_with("some_locator")
```

Comment:
Why would `click_element` be called in this test? 

I don't believe it is called in the `wait_for_and_mouse_over` method.

Reply:
this is wait_for_mouse_over_and_click

Reply:
Lol oops.